packages/fxa-auth-server/scripts/paypal-refund-fixer.ts

```diff
@@ -0,0 +1,182 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { AuthLogger } from 'fxa-auth-server/lib/types';
+import { ACTIVE_SUBSCRIPTION_STATUSES } from 'fxa-shared/subscriptions/stripe';
+import { StatsD } from 'hot-shots';
+import stripe from 'stripe';
+import Container from 'typedi';
+
+import { CurrencyHelper } from '../lib/payments/currencies';
+import { PayPalHelper } from '../lib/payments/paypal';
+import { PayPalClient } from '../lib/payments/paypal-client';
+import { STRIPE_INVOICE_METADATA, StripeHelper } from '../lib/payments/stripe';
+import { configureSentry } from '../lib/sentry';
+
+const config = require('../config').getProperties();
+
+class PayPalFixer {
+  private stripe: stripe;
+
+  constructor(
+    private log: AuthLogger,
+    private stripeHelper: StripeHelper,
+    private paypalHelper: PayPalHelper
+  ) {
+    this.stripe = (this.stripeHelper as any).stripe as stripe;
+  }
+
+  private activeSubscriptions(
+    subscriptions: stripe.Subscription[] | undefined
+  ) {
+    if (!subscriptions) {
+      return [];
+    }
+    return subscriptions.filter((sub) =>
+      ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
+    );
+  }
+
+  private async *nonRefundedPaypalInvoices(limit?: number) {
+    const startDate = new Date(2021, 3, 20).getTime() / 1000;
+    let count = 0;
+    for await (const invoice of this.stripe.invoices.list({
+      limit: 100,
+      status: 'paid',
+      collection_method: 'send_invoice',
+      created: { gt: startDate },
+    })) {
+      if (
+        !invoice.metadata![STRIPE_INVOICE_METADATA.PAYPAL_TRANSACTION_ID] ||
+        invoice.metadata![
+          STRIPE_INVOICE_METADATA.PAYPAL_REFUND_TRANSACTION_ID
+        ] ||
+        invoice.post_payment_credit_notes_amount === 0
+      ) {
+        continue;
+      }
+      yield invoice;
+      count++;
+      if (limit && count >= limit) {
+        break;
+      }
+    }
```

Comment:
If I'm reading the limits correctly, this will only process a max of 100 customers per execution.  Are we going to run this script multiple times then?

Reply:
That's not actually used, I can remove it. We actually get everyone the first time, because we're using async iteration, and its 100 per batch.